multidict/_multidict_views.c

```diff
@@ -0,0 +1,562 @@
+        keysview_repr_func, self, NULL);
+}
+
+static PyObject *
+multidict_keysview_isdisjoint(_Multidict_ViewObject *self, PyObject *other)
+{
+    return PyObject_CallFunctionObjArgs(
+        keysview_isdisjoint_func, self, other, NULL);
+}
+
+PyDoc_STRVAR(keysview_isdisjoint_doc,
+             "Return True if two sets have a null intersection.");
+
+static PyMethodDef multidict_keysview_methods[] = {
+    {
+        "isdisjoint",
+        (PyCFunction)multidict_keysview_isdisjoint,
+        METH_O,
+        keysview_isdisjoint_doc
+    },
+    {
+        NULL,
+        NULL
+    }   /* sentinel */
+};
+
+static int
+multidict_keysview_contains(_Multidict_ViewObject *self, PyObject *key)
+{
+    PyObject *impl = _PyObject_CallMethodId(self->md, &PyId_impl, NULL);
+    return pair_list_contains(impl, key);
+}
+
+static PySequenceMethods multidict_keysview_as_sequence = {
+    (lenfunc)multidict_view_len,               /* sq_length */
+    0,                                         /* sq_concat */
+    0,                                         /* sq_repeat */
+    0,                                         /* sq_item */
+    0,                                         /* sq_slice */
+    0,                                         /* sq_ass_item */
+    0,                                         /* sq_ass_slice */
+    (objobjproc)multidict_keysview_contains,   /* sq_contains */
+};
+
+static PyTypeObject multidict_keysview_type = {
+    PyVarObject_HEAD_INIT(DEFERRED_ADDRESS(&PyType_Type), 0)
+    "_KeysView",                                   /* tp_name */
+    sizeof(_Multidict_ViewObject),                 /* tp_basicsize */
+    0,                                             /* tp_itemsize */
+    (destructor)multidict_view_dealloc,            /* tp_dealloc */
+    0,                                             /* tp_print */
+    0,                                             /* tp_getattr */
+    0,                                             /* tp_setattr */
+    0,                                             /* tp_reserved */
+    (reprfunc)multidict_keysview_repr,             /* tp_repr */
+    &multidict_view_as_number,                     /* tp_as_number */
+    &multidict_keysview_as_sequence,               /* tp_as_sequence */
+    0,                                             /* tp_as_mapping */
+    0,                                             /* tp_hash */
+    0,                                             /* tp_call */
+    0,                                             /* tp_str */
+    0,                                             /* tp_getattro */
+    0,                                             /* tp_setattro */
+    0,                                             /* tp_as_buffer */
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,       /* tp_flags */
+    0,                                             /* tp_doc */
+    0,                                             /* tp_traverse */
```

Comment:
MultiDict stores any objects as values.
As a consequence, it can have circular deps to itself.
This implies `tp_traverse` and `tp_clear` slot implementation for all views and iterators.

Reply:
Ok, thx